easybuild/easyblocks/w/wps.py

```diff
@@ -93,8 +95,14 @@ def configure_step(self):
         # WRF dependency check
         wrf = get_software_root('WRF')
         if wrf:
-            majver = get_software_version('WRF').split('.')[0]
-            self.wrfdir = os.path.join(wrf, "WRFV%s" % majver)
+            wrfver = get_software_version('WRF')
+            majver = wrfver.split('.')[0]
+            if majver == '3':
+                self.wrfdir = os.path.join(wrf, "WRFV%s" % majver)
+            elif majver == '4':
+                self.wrfdir = os.path.join(wrf, "WRF-%s" % wrfver)
+            else:
+                raise EasyBuildError("Unknown WRF version: %s" % wrfver)
```

Comment:
I would make this less strict, and assume that ‘WRF-‘ is used from version 4.x onwards.

Even better would be to employ a slightly more dynamic approach to finding the subdir, e.g. go with the subdirectory that starts with ‘WRF’ if there’s only one, and error out otherwise.

Reply:
not sure if going the dynamic route is worth it, given potential future changes in WRF layout.

Reply:
At least then there's a chance that this will work, so it's worth a shot imho.

Reply:
The logical place to put this would be the new `det_wrf_subdir` function.  However, then, we would need to check for existence of the directory in that subdir.  In the wrf easyblock, the function gets called from `__init__()`.

I'm not sure about the internals of EasyBuild; however I suspect that by the time the easyblock's `__init__` is being run, the sources have not been unpacked yet, so we cannot implement the dynamic route there.

Would it be safe to call the `det_wrf_subdir` from the `configure_step()`? That way, we could rely on the sources being unpacked, and could implement the dynamic check without code duplication.

Reply:
It depends where the result from `det_wrf_subdir` is used. We should try and make sure that things work properly when `--module-only` is used (which skips the `configure` step entirely, but doesn't skip `sanity_check_step`).

One downside of checking for a particular subdir is that it'll interfere with `-x` (i.e. it may result in an error because no files are present whatsoever under `-x`). That's something we should try and avoid...

So, with all that in mind, I'd keep it the way it is now in `det_wrf_subdir`: use `WRF-<version>` for newer versions. If needed, it'll be easier to update that for future versions.

Reply:
agreed.